sycl/include/sycl/ext/oneapi/experimental/device_architecture.hpp

```diff
@@ -1009,6 +1012,163 @@ template <bool MakeCall> class if_architecture_helper {
 
 namespace ext::oneapi::experimental {
 
+#ifdef SYCL_EXT_ONEAPI_DEVICE_ARCHITECTURE_NEW_DESIGN_IMPL
```

Comment:
I would suggest to reduce the scope of `#ifdef` to only have user-facing `if_architecture_is` guarded by it. 

I assume that mere presence of `detail` namespace and definitions within it doesn't break anything? By making it available by default, you will also get some testing for it